src/coreclr/src/vm/dumpdiagnosticprotocolhelper.cpp

```diff
@@ -55,17 +55,19 @@ const GenerateCoreDumpCommandPayload* GenerateCoreDumpCommandPayload::TryParse(B
         return nullptr;
     }
 
-    payload->incomingBuffer = lpBuffer;
-    uint8_t* pBufferCursor = payload->incomingBuffer;
+    uint8_t* pBufferCursor = lpBuffer;
```

Comment:
Does `GenerateCoreDumpCommandPayload*` above need to be `NewArrayHolder` now?

Reply:
The user is expected to handle the memory `new`'d here.  Down on line 88, the `GenerateCoreDumpPayload` is put into a `NewHolder`.

Reply:
It is a good practice to assign a newly allocated memory to a holder immediately. That way, you would not need to delete it at line 65 and it is also future proof when someone might add another return path to this function. The final return in this function would then use payload->Extract() or just call SuppressRelease() before the return.